lib/replica_pools/connection_proxy.rb

```diff
@@ -33,10 +33,16 @@ def initialize(leader, pools)
       # this ivar is for ConnectionAdapter compatibility
       # some gems (e.g. newrelic_rpm) will actually use
       # instance_variable_get(:@config) to find it.
-      if ActiveRecord::VERSION::MAJOR >= 7
-        @config = current.connection_db_config
+      @config = current.send(ReplicaPools::ConnectionProxy.get_connection_config_method_name)
+    end
+
+    def self.get_connection_config_method_name
+      # 6.1 supports current.connection_config
+      # but warns of impending deprecation in 6.2
+      if [ActiveRecord::VERSION::MAJOR, ActiveRecord::VERSION::MINOR].join(".").to_f >= 6.1
```

Comment:
could also use `ActiveRecord::VERSION::STRING.to_f` instead of manually concatenating major/minor here!

```
[30] pry(main)> ActiveRecord::VERSION::STRING
=> "6.1.0"
[31] pry(main)> ActiveRecord::VERSION::STRING.to_f
=> 6.1

```